cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/dataset/SQLDataset.java

```diff
@@ -14,14 +14,16 @@
  * the License.
  */
 
-package io.cdap.cdap.etl.engine;
+package io.cdap.cdap.etl.api.engine.sql.dataset;
+
+import io.cdap.cdap.api.data.schema.Schema;
 
 /**
- * Status for SQL Engine Jobs.
+ * Represents a dataset that lives in a SQL engine outside of spark
+ * @param <T> the type for the records stored in the SQL Engine.
  */
-public enum SQLEngineJobStatus {
-  CREATED,
-  RUNNING,
-  COMPLETED,
-  FAILED
+public interface SQLDataset<T> {
```

Comment:
I don't think you need T here. If dataset is never materialized on Java side, T is not defined, it's inside BQ. Also it's always StructuredRecord when on Java side, is not it?

Reply:
Fair point, I just wanted to make it clear that a SQLDataset represents an underlying "something". We currently only use `StructuredRecord` but I see no reason to limit this.

Reply:
Yes, a SQLDataset represents a collection of records that exists in the SQL engine. This collection of records represents either records that were pushed, or the result of an operation in the SQL engine (currently, only Join results).

I'm not thinking about combining stages just yet. This simply allows me to represent the distinct stages that exist in the SQL engine (e.g. tableA, tableA, tableAjoinB, etc). and allows me to "track" which datasets exists in the engine.

Reply:
That's fine in the implementation. But do you need it in the interface?

Reply:
I have removed the type hint from the interface.